### DISCUSSION TITLE
VueJS+Vite example (using dev-server / build step)

### DISCUSSION THREAD

**Discussion by Moartl31** [TotallyInformation/node-red-contrib-uibuilder#211] [Q&A] [answered]
In the uibuilder docs it is very well documented how to use a development server and which temporary changes to the code are necessary. Only Svelte works "out of the box".

It is also mentioned that these manual changes could be automated using the build tool (custom config I assume). As VueJS+Vite is a very popular combo, it would be cool to have a working template with a custom vite-config which "just works" without having to change certain things temporarily. At least it would be good to know if it is at all possible to adjust the Vite-Config accordingly.

**Comment by TotallyInformation** [comment, marked as answer]
Hi, thanks for reaching out and sorry for the delayed response.

I would love to have a vite config that just works! If you can work it out, please do share so that I can put it into the docs.

Unfortunately, I don't use VueJS any more so while I can get the basic build config done, I don't have the knowledge to know how to configure Vite to avoid the need for changing things in your front-end code. I _believe_ that it might be possible and I've had a few goes at it but I've not managed to achieve it.

I think that Svelte works only because you can continue to use Node-RED as the server but Svelte's dev server still makes a connection somehow and so manages the reload of the page after saving a change to the source.
